pkg/keyspace/keyspace.go

```diff
@@ -54,13 +56,24 @@ func GetKeyspaceNameBySettings() (keyspaceName string) {
 	return keyspaceName
 }
 
+var keyspaceNameBytes []byte
+var genKeyspaceNameOnce sync.Once
+
 // GetKeyspaceNameBytesBySettings is used to get keyspace name setting as a byte slice.
 func GetKeyspaceNameBytesBySettings() []byte {
-	keyspaceName := config.GetGlobalKeyspaceName()
-	if len(keyspaceName) == 0 {
-		return nil
-	}
-	return []byte(keyspaceName)
+	genKeyspaceNameOnce.Do(func() {
+		if !kerneltype.IsNextGen() {
+			return
+		}
+
+		keyspaceName := config.GetGlobalKeyspaceName()
+		if len(keyspaceName) == 0 {
+			return
+		}
+		keyspaceNameBytes = []byte(keyspaceName)
+		return
+	})
```

Comment:
Benchmark result:
Before this change: 
`BenchmarkGetKeyspaceNameBytesBySettings-8   	58244062	        32.69 ns/op	      24 B/op	       1 allocs/op`
After this change : 
`BenchmarkGetKeyspaceNameBytesBySettings-8   	593358097	         2.053 ns/op	       0 B/op	       0 allocs/op`